### PR TITLE
Refactor access to trainer attributes in LightningModule

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Implements read-only passthrough for the attributes that are exclusively handled by the Trainer. This avoids accidental divergence of state. Additionally, just to be safe we make the trainer a week-ref.

Refactor is also partially necessary for #5616 

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

File: tests/core/test_lightning_module.py [hunk 1]
```diff
@@ -11,16 +11,68 @@
...
 import pytest
 from torch.optim import Adam, SGD
 
 from pytorch_lightning import Trainer
+from pytorch_lightning.loggers import TensorBoardLogger
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
 from tests.base import BoringModel
 
 
+def test_property_current_epoch():
```
Comment (Borda) [inline tests/core/test_lightning_module.py:25]: lest move it to separate file `tests/core/test_module_attribites.py`?

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -107,6 +107,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Changed the default value for the `progress_bar_refresh_rate` Trainer argument in Google COLAB notebooks to 20 ([#5516](https://github.com/PyTorchLightning/pytorch-lightning/pull/5516))
 
 
+- Made `LightningModule.global_rank`, `LightningModule.local_rank` and `LightningModule.logger` read-only properties ([#5730](https://github.com/PyTorchLightning/pytorch-lightning/pull/5730))
+
+
+- Changed trainer-related attributes in LightningModule as read-only property: `global_rank`, `local_rank`, `logger` ([#5730](https://github.com/PyTorchLightning/pytorch-lightning/pull/5730))
```
Comment (Borda) [inline CHANGELOG.md:113]:
```suggestion
- Made `LightningModule.global_rank`, `LightningModule.local_rank` and `LightningModule.logger` read-only properties ([#5730](https://github.com/PyTorchLightning/pytorch-lightning/pull/5730))
```

File: tests/core/test_lightning_module.py (hunk 1 above)
Comment (awaelchli) [reply tests/core/test_lightning_module.py:25]: hmm, not sure. Isn't it more logical that unit tests for pl/core/lightning_module go to tests/core/test_lightning?

Comment (Borda) [reply tests/core/test_lightning_module.py:25]: just try to keep the file light :]

**Review by tchaton — APPROVED** [review]
LGTM !
